src/components/Common/Fields/TextField/TextField.tsx

```diff
@@ -0,0 +1,39 @@
+import type React from 'react'
+import type { RegisterOptions } from 'react-hook-form'
+import { useFormContext, useController } from 'react-hook-form'
+import {
+  TextField as TextFieldUi,
+  type TextFieldProps as TextFieldUiProps,
+} from '../../UI/TextField'
+
+type TextFieldProps = TextFieldUiProps & {
+  rules?: RegisterOptions
+  defaultValue?: string
+  name: string
+}
+
+export const TextField: React.FC<TextFieldProps> = ({
+  rules,
+  defaultValue,
+  name,
+  errorMessage,
+  ...props
+}: TextFieldProps) => {
+  const { control } = useFormContext()
+  const { field, fieldState } = useController({
```

Comment:
Can we somehow get the controller code into a reusable component? We can chat more about this too.

Reply:
Feels like HookField like I sorta had earlier might be able to handle this? But I like this separation overall across all the fields.

Reply:
I updated to try and place some of the common mappings in a hook. The idea would be to effectively just abstract away the direct calls to hook form and return values that were readily mapped to props in our namespace